public/main.js

```diff
@@ -1,29 +1,156 @@
+  stringArray.push(stringNumber)
+  const slicedArray = stringArray[0].slice(0, 5)
+  return slicedArray
+}
+
+function toUtcStringSliced(dateObj) {
+  const utcDate = (dateObj).toString()
+  const date = new Date(utcDate)
+  const newDate = new Date(date.getFullYear(), date.getMonth(), date.getDate())
+  const newerDate = newDate.toUTCString()
+  const newestDate = newerDate.split(' ').slice(0, 4).join(' ')
+  const twelveDate = formatTime(date)
+  const utcStringSliced = newestDate + '; ' + twelveDate
+  return utcStringSliced
+}
+
+function createATagButton(url) {
+  const aTag = document.createElement('a')
+  aTag.setAttribute('class', 'ui positive button')
+  aTag.href = url
+  aTag.textContent = 'Buy Tickets'
+  return aTag
+}
+
+function formatTime(dateObj) {
+  var hour = dateObj.getHours()
+  var minute = dateObj.getMinutes()
+  var amPM = (hour > 11) ? 'pm' : 'am'
+  if (hour > 12) {
+    hour -= 12
+  }
+  else if (hour === 0) {
+    hour = '12'
+  }
+  if (minute < 10) {
+    minute = '0' + minute
+  }
+  return hour + ':' + minute + amPM
+}
+
+function sortExpRating(object) {
+  object.sort((a, b) => {
+    return parseFloat(b.experienceRating) - parseFloat(a.experienceRating)
+  })
+}
+
+function sortDataDescending(object, key) {
+  object.sort((a, b) => {
+    return parseFloat(b[key]) - parseFloat(a[key])
+  })
+}
+
+function sortDataAscending(object, key) {
+  object.sort((a, b) => {
+    return parseFloat(a[key]) - parseFloat(b[key])
+  })
+}
+
+function sortDateAscending(object) {
+  object.sort((a, b) => {
+    return new Date(a.date) - new Date(b.date)
+  })
+}
+
+function sortDateDescending(object) {
+  object.sort((a, b) => {
+    return new Date(b.date) - new Date(a.date)
+  })
+}
+
+function renderTableData(teamArray) {
+  const tableBody = document.querySelector('tbody')
+  tableBody.innerHTML = ''
+  teamArray.forEach(object => {
+    const tableRow = document.createElement('tr')
+    for (let key in object) {
+      const tableData = document.createElement('td')
+      if (key === 'url') {
+        const aTag = createATagButton(object.url)
+        tableData.appendChild(aTag)
+      }
+      else if (key === 'experienceRating') {
+        const rating = (object[key] * 100)
+        const slicedArray = toSlicedArray(rating)
+        tableData.textContent = slicedArray
+      }
+      else if (key === 'date') {
+        const newestDate = toUtcStringSliced(object.date)
+        tableData.textContent = newestDate
+      }
+      else {
+        tableData.textContent = object[key]
+      }
+      tableRow.appendChild(tableData)
+    }
+    tableBody.appendChild(tableRow)
+  })
+  table.appendChild(tableBody)
+}
+
+function toggleData(object, key) {
+  const header = document.querySelector('thead')
+  let clickCount = 0
+  header.addEventListener('click', (event) => {
+    const $targetDiv = event.target
+    const dataAttr = $targetDiv.getAttribute('data-key')
+    if (dataAttr === key & clickCount % 2 === 0) {
```

Comment:
instead of keeping track of `clickCount`, you can add a class or data attribute to `event.target` so you can know what state it is in.